### PR TITLE
Feat/device authorization grant

### PR DESCRIPTION
**Pull request by pedrule**
## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation within the code base (if appropriate).

### CODE REVIEW

File: device_authorize_request_handler.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+ import (
+	 "context"
+	 "net/http"
+	 "strings"
+ 
+	 "github.com/ory/x/errorsx"
+ )
+ 
+ func (f *Fosite) NewDeviceAuthorizeRequest(ctx context.Context, r *http.Request) (DeviceAuthorizeResponder, error) {
```
Comment (james-d-elliott) [inline device_authorize_request_handler.go:32]:
```suggestion
 func (f *Fosite) NewDeviceAuthorizeRequest(ctx context.Context, r *http.Request) (DeviceAuthorizeRequester, error) {
```

Comment (james-d-elliott) [reply device_authorize_request_handler.go:32]: I think this is the appropriate change as per the test suite.

**Review by james-d-elliott — CHANGES_REQUESTED** [review]
Think my review addresses one of the issues, not sure about the rest here: https://app.circleci.com/pipelines/github/ory/fosite/717/workflows/ad2e8279-e2a8-4e6b-8452-68d1556330a4/jobs/2287

**Comment by pedrule** [comment]
No more needed, a new one is coming

**Pull request closed without merging**